### PR TITLE
Merge 0.4 to 0.5

### PR DESCRIPTION
**Pull request by KalervoHyyppa**
## Description

This merges rc-0.4 into dev (rc-0.5)

Fixes # (issue)

[RIBN-782](https://topl.atlassian.net/jira/software/c/projects/RIBN/boards/5?modal=detail&selectedIssue=RIBN-782)

## Checklist:

-   [x] My code follows the style guidelines of this project
-   [x] I have performed a self-review of my own code
-   [x] I have commented my code, particularly in hard-to-understand areas
-   [x] I have made corresponding changes to the documentation
-   [x] My changes generate no new warnings
-   [x] I have added tests that prove my fix is effective or that my feature works
-   [x] New and existing unit tests pass locally with my changes
-   [x] Any dependent changes have been merged and published in downstream modules
-   [x] I have checked my code and corrected any misspellings

[RIBN-782]: https://topl.atlassian.net/browse/RIBN-782?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: pubspec.yaml [hunk 1]
```diff
@@ -101,6 +101,7 @@ dev_dependencies:
 #  custom_lint: ^0.2.5
 #  riverpod_lint: ^1.0.1
 
+  built_value_generator: ^8.1.2
```
Comment (ultimaterex) [inline pubspec.yaml:104]: What are we using built value for that freezed doesn't do for us?

**Review by ultimaterex — CHANGES_REQUESTED** [review]

File: lib/main.dart [hunk 2]
```diff
@@ -30,6 +30,8 @@ import 'package:ribn/providers/store_provider.dart';
 import 'package:ribn/redux.dart';
 import 'package:ribn/router/root_router.dart';
 
+// import 'package:ribn/platform/web/wallet.dart';
```
[the remark this answers is not in the document]
Comment (KalervoHyyppa) [reply lib/main.dart:33]: Looks like it was added by Mustafa 1.5 years ago, I'll remove

File: pubspec.yaml (hunk 1 above)
Comment (KalervoHyyppa) [reply pubspec.yaml:104]: Not sure what this package does, must have been left over from another 0.4 PR; removing.

**Review by ultimaterex — APPROVED** [review]

**Review by scasplte2 — APPROVED** [review]
🔥

**Pull request merged**